CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -2441,7 +2441,7 @@ protected function doMembershipProcessing($contactID, $membershipParams, $premiu
         $membershipLineItems = [];
         foreach ($this->_values['fee'] as $key => $feeValues) {
           if ($feeValues['name'] == 'membership_amount') {
-            $fieldId = $this->_params['price_' . $key];
+            $fieldId = $this->_submitValues['price_' . $key];
```

Comment:
@eileenmcnaughton @mattwire `$this->_submitValues` does not have any values filled in, so this does not assign anything to `$fieldId` after the confirm page is submitted. I get following DB error 

>Database Error Code: Field 'qty' doesn't have a default value, 1364

which can be replicated by submitting a contribution page with both contribution and membership + enable `Separate Membership Payment` checkbox. The following notice error is also displayed after applying the patch -

>Notice: Undefined index: price_5 in CRM_Contribute_Form_Contribution_Confirm->doMembershipProcessing() (line 2445 of /Users/jitendra/src/civicrm/CRM/Contribute/Form/Contribution/Confirm.php).
>Notice: Undefined index: in CRM_Contribute_Form_Contribution_Confirm->doMembershipProcessing() (line 2446 of /Users/jitendra/src/civicrm/CRM/Contribute/Form/Contribution/Confirm.php).